tests/experimental/test_mcmc.py

```diff
@@ -202,16 +242,54 @@ def test_checkpointing(sampler: cuqi.experimental.mcmc.SamplerNew):
 
     # Do some more samples from pre-defined rng state
     np.random.seed(0)
-    samples1 = sampler.sample(100).get_samples().samples
+    samples1 = sampler.warmup(50).sample(50).get_samples().samples
 
     # Now load the checkpoint on completely fresh sampler not even with target
     sampler_fresh = sampler.__class__(sampler.target) # In principle init with no arguments. Now still with target
     sampler_fresh.load_checkpoint('checkpoint.pickle')
 
     # Do some more samples from pre-defined rng state
     np.random.seed(0)
-    samples2 = sampler_fresh.sample(100).get_samples().samples[...,1:] # TODO. This needs to be fixed..
+    samples2 = sampler_fresh.warmup(50).sample(50).get_samples().samples[...,1:] # TODO. This needs to be fixed.. We should likely not store initial point in _samples
 
     # Check that the samples are the same
     assert np.allclose(samples1, samples2), f"Samples1: {samples1}\nSamples2: {samples2}"
 
+
+@pytest.mark.parametrize("sampler", checkpoint_targets)
```

Comment:
we can probably make these 2 tests for state keys and history keys general to all samplers not only those that are tested for checkpointing.

Reply:
Done